shared/src/main/scala/fetch.scala

```diff
@@ -189,8 +189,12 @@ object `package` {
         }
       } yield result)
 
+    // todo: can be tail recursive?
     def tailRecM[A, B](a: A)(f: A => Fetch[Either[A, B]]): Fetch[B] =
-      ???
+      f(a).flatMap(_ match {
+        case Left(a) => tailRecM(a)(f)
+        case Right(b) => pure(b)
+      })
```

Comment:
it would be possible to express this with `fold` (I guess you'd need the `cats.syntax.either._`)?

```scala
f(a).flatMap(_.fold(tailRecM(_)(f), pure(_))
```

Reply:
I will update it. However, I'm not sure if we can write `tailRecM` as a tail recursive method since we need `.flatMap`.